.github/actions/detect-file-changes/detect-file-changes.js

```diff
@@ -0,0 +1,246 @@
+// Lightweight in-repository file-change filter for GitHub pull request workflows.
+//
+// It performs the following:
+//   1. Reads a YAML-like input containing glob patterns.
+//      - Supports negation via leading '!'
+//      - Supports wildcards: *, **, ?
+//   2. Fetches changed files from the current Pull Request using the GitHub API.
+//   3. Determines whether any changed file matches the patterns.
+//   4. Outputs `changed=true|false` via GITHUB_OUTPUT.
+//
+// Example usage in a workflow:
+//
+//   steps:
```

Comment:
Can we add coding formatting here? 
```suggestion
//   ```yaml
//   steps:
```